rfd/0185-k8s-access-non-cert-routing.md

```diff
@@ -0,0 +1,247 @@
+
+Within the Kube Forwarder, new route handlers would be added for all HTTP
+methods for `/v1/teleport/:teleportCluster/:kubeCluster/*`. These route handlers
+would parse the `teleportCluster` and `kubeCluster` from the URL, strip off the
+prefix, and then pass the request through to the original handler - with the
+additional context of the desired Teleport and Kubernetes cluster. This would
+take higher precedence than attributes contained within the certificate.
+
+Teleport clients would no longer need to call `GenerateUserCerts` with a
+`KubernetesCluster` to produce a certificate explicitly for access to
+Kubernetes. When producing a kubeconfig, the client should use the user's 
+standard identity.
+
+To configure Kubernetes clients to work with this, the Kubeconfig generated by
+Teleport would be updated to include this base path within the
+`clusters[name].cluster.server` field:
+
+```yaml
+apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority-data: ...omitted... # No-change
+    # Before:
+    # server: https://example.teleport.sh/
+    # After:
+    server: https://example.teleport.sh/v1/teleport/base64-teleport-cluster-name/base64-k8s-cluster-name
+    tls-server-name: kube-teleport-proxy-alpn.example.teleport.sh # No-change
+  name: example-example
+```
+
+#### Backwards/Forwards Compatibility
+
+By retaining the existing behaviour and endpoints within the Kube Forwarder,
+older clients using a kubeconfig and certificate generated with the
+`KubernetesCluster` attribute will continue to function.
+
+- VX.Y.0:
+  - Introduce new Kube Forwarder behaviour.
+  - Introduce opt-in `kubernetes/v2` `tbot` output. This will allow
+    users to opt-in to the new behaviour once they are satisfied that their
+    Auth Server, Proxy and Kubernetes Agents have been upgraded to the required
+    version.
+- V(X+1).0.0:
+  - Switch `tsh` to new behaviour. No longer requests the signing of a
+    certificate with the `KubernetesCluster` attribute and uses standard user
+    identity to authenticate.
+  - Switch `tbot` `kubernetes/v1` output to use `kubernetes/v2` implementation
+    under the hood.
+- V(X+2).0.0:
+  - It is now "safe" to remove the old attribute based routing from the Kube
+    Forwarder. We may wish to delay this further if we have concerns about
+    older client compatability.
+
+#### Client Compatability
+
+Kubernetes API clients have been known to vary greatly in their support for 
+more unusual configuration. We have witnessed this in the past when it came to
+supporting the ability to explicitly provide the SNI to use when making
+requests.
+
+Therefore, we should ensure that Kubernetes clients and SDKs do support
+providing a base path as part of the server address (e.g they do not strip off
+the path and only use the hostname/port).
+
+The key customer we are working with has specified that they use the following
+Kubernetes clients:
+
+- `kubectl`
+- Python Kubernetes SDK
```

Comment:
It's worth checking javascript client as we have some that also use it for automation.